src/stream_realsense.cpp

```diff
@@ -187,18 +190,16 @@ StreamRealSense::create_gstreamer_pipeline(std::map<std::string, std::string> &p
         rs_set_device_option(ctx->dev, RS_OPTION_R200_LR_AUTO_EXPOSURE_ENABLED, 1, NULL);
 
     /* gstreamer */
-    GError *error = nullptr;
-    std::string pipeline_str = create_pipeline(params);
-    GstElement *pipeline = gst_parse_launch(pipeline_str.c_str(), &error);
+    pipeline = gst_parse_launch(create_pipeline(params).c_str(), &error);
     if (!pipeline) {
         log_error("Error processing pipeline for RealSense stream device: %s\n",
                   error ? error->message : "unknown error");
         g_clear_error(&error);
 
-        return nullptr;
+        goto error_rs;
     }
 
-    GstElement *appsrc = gst_bin_get_by_name(GST_BIN(pipeline), "mysource");
+    appsrc = gst_bin_get_by_name(GST_BIN(pipeline), "mysource");
```

Comment:
This function increases the reference count of the object. Once done, the caller need to decrease the reference count on the object by calling gst_object_unref().